CMakeLists.txt

```diff
@@ -70,6 +75,17 @@ if (USE_OPENMP)
   endif ()
 endif ()
 
+set(LIBCMAES_DEPS "")
+
+if (USE_TBB)
+  find_package (TBB)
+  if (TBB_FOUND)
+    add_definitions (-DUSE_TBB)

```

Comment:
maybe you could add a per-target include

Reply:
> maybe you could add a per-target compile definition

You mean to produce 2 library targets:

- one without TBB
- and one with TBB (if TBB is available)

Otherwise, I do not really see what would be the difference (other than just using a different CMake commad); the library has quite a few targets and `add_definitions` is easier to use in this case (and does not really pollute anything, does it?).

Reply:
no, I mean define the symbol just for this target, ie not define it for tests, python, that's cleaner

Reply:
> no, I mean define the symbol just for this target, ie not define it for tests, python, that's cleaner

But we do want it for the tests (I am not sure for python); I have it in my todo list to make a new PR with additional setups in travis to check for the usage of TBB.